contracts/SimpleOperations.sol

```diff
@@ -45,106 +46,210 @@ contract SimpleOperations is Operations {
+	}
 
 	// Functions for client owners
 
-	function setClientOwner(address _newOwner) public only_client_owner {
+	function setClientOwner(address _newOwner)
+		public
+		only_client_owner
+		not_client_owner(_newOwner)
+	{
 		var newClient = clientOwner[msg.sender];
 		clientOwner[msg.sender] = bytes32(0);
 		clientOwner[_newOwner] = newClient;
 		client[newClient].owner = _newOwner;
 		ClientOwnerChanged(newClient, msg.sender, _newOwner);
 	}
 
-	function addRelease(bytes32 _release, uint32 _forkBlock, uint8 _track, uint24 _semver, bool _critical) public only_client_owner {
+	/// Adds a release which is identified by `_release`. When a release is succesfully added, a
+	/// `ReleaseAdded` event is emitted. The same release can be added multiple times (with
+	/// different parameters), therefore the `ReleaseAdded` event can be emitted multiple times for
+	/// the same release.
+	function addRelease(
+		bytes32 _release,
+		uint32 _forkBlock,
+		uint8 _track,
+		uint24 _semver,
+		bool _critical
+	)
+		public
+		only_client_owner
+	{
 		var newClient = clientOwner[msg.sender];
-		client[newClient].release[_release] = Release(_forkBlock, _track, _semver, _critical);
+		client[newClient].release[_release] = Release(
+			_forkBlock,
+			_track,
+			_semver,
+			_critical
+		);
 		client[newClient].current[_track] = _release;
-		ReleaseAdded(newClient, _forkBlock, _release, _track, _semver, _critical);
+		ReleaseAdded(
+			newClient,
+			_forkBlock,
+			_release,
+			_track,
+			_semver,
+			_critical
+		);
 	}
 
-	function addChecksum(bytes32 _release, bytes32 _platform, bytes32 _checksum) public only_client_owner {
+	/// Adds a `_checksum` for a `_release` binary targeting a given `_platform`. When a checksum is
+	/// succesfully added, a `ChecksumAdded` event is emitted. The same checksum can be added
+	/// multiple times (with different parameters), therefore the `ChecksumAdded` event can be
+	/// emitted multiple times for the same checksum.
+	function addChecksum(bytes32 _release, bytes32 _platform, bytes32 _checksum)
+		public
+		only_client_owner
+	{
 		var newClient = clientOwner[msg.sender];
 		client[newClient].build[_checksum] = Build(_release, _platform);
 		client[newClient].release[_release].checksum[_platform] = _checksum;
-		ChecksumAdded(newClient, _release, _platform, _checksum);
+		ChecksumAdded(
+			newClient,
+			_release,
+			_platform,
+			_checksum
+		);
 	}
 
 	// Admin functions
 
-	function addClient(bytes32 _client, address _owner) public only_owner {
+	function addClient(bytes32 _client, address _owner)
+		public
+		only_owner
+		not_client_owner(_owner)
+	{
+		// we can't add a client that already exists
+		require(client[_client].owner == 0);
 		client[_client].owner = _owner;
 		clientOwner[_owner] = _client;
 		ClientAdded(_client, _owner);
 	}
 
-	function removeClient(bytes32 _client) public only_owner {
+	function removeClient(bytes32 _client)
+		public
+		only_owner
+	{
 		resetClientOwner(_client, 0);
```

Comment:
This won't work, after this function is called `0` becomes an owner of `_client` and then you cannot remove any other client.

Reply:
Why do we need so many events and functions anyway? Why not just `setClient(_client, _owner)` (covers adding and changing) and `removeClient(_client, _owner)` that get's rid of all traces of particular client (TBH I'm not sure if it's even needed after we removed the fork voting stuff)?